x-pack/plugins/cases/server/services/cases/transform.ts

```diff
@@ -174,6 +178,9 @@ export function transformSavedObjectToExternalModel(
     SEVERITY_ESMODEL_TO_EXTERNAL[caseSavedObjectAttributes.severity] ?? CaseSeverity.LOW;
   const status = STATUS_ESMODEL_TO_EXTERNAL[caseSavedObjectAttributes.status] ?? CaseStatuses.open;
   const category = !caseSavedObjectAttributes.category ? null : caseSavedObjectAttributes.category;
+  const customFields = !caseSavedObjectAttributes.customFields
+    ? []
+    : (caseSavedObjectAttributes.customFields as unknown as CaseCustomFields);
```

Comment:
Now that the types are correct I think the `as unknown` is not needed. You can also do `as CaseTransformedAttributes['customFields']` instead of `CaseCustomFields`.